predicate/predicate_slots.go

```diff
@@ -5,19 +5,21 @@ package predicate
 
 import (
 	"github.com/ava-labs/coreth/core/types"
-	"github.com/ava-labs/coreth/params"
 	"github.com/ava-labs/coreth/utils"
 	"github.com/ethereum/go-ethereum/common"
 )
 
+type predicaters interface {
+	PredicaterExists(common.Address) bool
+}
+
 // PreparePredicateStorageSlots populates the the predicate storage slots of a transaction's access list
 // Note: if an address is specified multiple times in the access list, each storage slot for that address is
 // appended to a slice of byte slices. Each byte slice represents a predicate, making it a slice of predicates
 // for each access list address, and every predicate in the slice goes through verification.
-func PreparePredicateStorageSlots(rules params.Rules, list types.AccessList) map[common.Address][][]byte {
+func PreparePredicateStorageSlots(rules predicaters, list types.AccessList) map[common.Address][][]byte {
```

Comment:
hmm could the same thing be used in CheckPredicates (/core/evm.go) if importing params also makes it complicated there?

Reply:
the core package already has to import params so it's not an issue there.